tests/api/test_links_with_root_path.py

```diff
@@ -0,0 +1,133 @@
+import pytest
+from httpx import ASGITransport, AsyncClient
+
+from stac_fastapi.pgstac.db import close_db_connection, connect_to_db
+
+root_path_value = "/stac/v1"
+
+# Append the root path to the base URL, this is key to reproducing the issue where the root path appears twice in some links
+base_url = f"http://api.acme.com{root_path_value}"
+
+
+@pytest.fixture(scope="function")
+async def app_with_root_path(database, monkeypatch):
+    """
+    Provides the global stac_fastapi.pgstac.app.app instance, configured with a
+    specific ROOT_PATH environment variable and connected to the test database.
+    """
+
+    monkeypatch.setenv("ROOT_PATH", root_path_value)
+    monkeypatch.setenv("PGUSER", database.user)
+    monkeypatch.setenv("PGPASSWORD", database.password)
+    monkeypatch.setenv("PGHOST", database.host)
+    monkeypatch.setenv("PGPORT", str(database.port))
+    monkeypatch.setenv("PGDATABASE", database.dbname)
+    monkeypatch.setenv("ENABLE_TRANSACTIONS_EXTENSIONS", "TRUE")
+
+    from stac_fastapi.pgstac.app import app, with_transactions
+
+    # Ensure the app's root_path is configured as expected
+    assert (
+        app.root_path == root_path_value
+    ), f"app_with_root_path fixture: app.root_path is '{app.root_path}', expected '{root_path_value}'"
+
+    await connect_to_db(app, add_write_connection_pool=with_transactions)
+    yield app
+    await close_db_connection(app)
+
+
+@pytest.fixture(scope="function")
+async def client_with_root_path(app_with_root_path):
+    async with AsyncClient(
+        transport=ASGITransport(app=app_with_root_path),
+        base_url=base_url,
```

Comment:
with root_path appended

Reply:
But the `root_path` shouldn't be appended to the base url, it should be provided with the additional `root-path` parameter 

https://github.com/stac-utils/stac-fastapi-pgstac/blob/main/tests/api/test_links.py#L58-L61

Reply:
Sorry, but I still don't understand how the client setup is affecting the server response. We are using httpx.AsynClient in the test for convenience, but I think it would happen with `curl`, `urllib`, `requests`, etc.  
I mean, isn't the client, behind the scenes, just performing a GET http request ?

Also I could not find a `root_path` parameter in the `AsyncClient` constructor:
https://github.com/encode/httpx/blob/4fb9528c2f5ac000441c3634d297e77da23067cd/httpx/_client.py#L1307